### ISSUE TITLE
ros2 service test_list_count failing

### ISSUE THREAD

**Issue by rotu** [ros2/ros2cli#466]
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu Focal
- Installation type:
  - source
- Version or commit hash:
  - master
- DDS implementation:
  - rmw_cyclonedds, rmw_fastrtps_cpp, rmw_connext_cpp
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue

see CI:
https://ci.ros2.org/view/nightly/job/nightly_linux_release/1486/testReport/ros2service.src.ros2.ros2cli.ros2service.test/test_cli/test_cli/

or locally:
```
colcon test --packages-select ros2service
```

#### Expected behavior

Tests pass

#### Actual behavior

Tests fail with a vague error message:

"some test cases have failed: test_cli[rmw_cyclonedds_cpp] failed at TestROS2ServiceCLI.test_list_count[rmw_cyclonedds_cpp] test_cli[rmw_fastrtps_cpp] failed at TestROS2ServiceCLI.test_list_count[rmw_fastrtps_cpp] test_cli[rmw_connext_cpp] failed at TestROS2ServiceCLI.test_list_count[rmw_connext_cpp]"

#### Additional information

Looking at the entire test output, this is probably due to a hack based on `launch_ros` implementation details which may have changed. In particular, 6 extra internal nodes may no longer exist.

```
======================================================================
FAIL: test_list_count[rmw_connext_cpp] (test_cli.TestROS2ServiceCLI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-agent/workspace/nightly_linux_release/ws/install/launch_testing/lib/python3.6/site-packages/launch_testing/markers.py", line 57, in _wrapper
    return func(self, *args, **kwargs)
  File "/home/jenkins-agent/workspace/nightly_linux_release/ws/src/ros2/ros2cli/ros2service/test/test_cli.py", line 195, in test_list_count
    assert int(output_lines[0]) == 7 + 6  # cope with launch_ros internal node.
AssertionError: assert 7 == (7 + 6)
 +  where 7 = int('7')
```

**Comment by ros-discourse** [comment]
This issue has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-2020-03-18/13313/1

**Comment by sloretz** [comment]
Fixed in #474
